pkg/dcmtime/date.go

```diff
@@ -0,0 +1,189 @@
+package dcmtime
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+)
+
+// Date holds data for a parsed DICOM date (DA value).
+type Date struct {
+	// Time is the underlying time.Time value.
+	Time time.Time
+
+	// Precision with which the raw DA value was stored. For instance, a Date value
+	// with a precision of PrecisionYear ONLY stored the year.
+	Precision PrecisionLevel
+
+	// IsNEMA will be true if this is a legacy NEMA-300 formatted value (1999.01.03).
+	IsNEMA bool
+}
+
+// DCM converts time.Time value to dicom DA string. Values are truncated to the
+// Date.Precision value.
+//
+// NOTE: Time zones are ignored in this operation, as DA does not support encoding them.
+// Make sure values are converted to UTC before passing if that is the desired output.
+func (da Date) DCM() string {
+	year, month, day := da.Time.Date()
+	builder := new(strings.Builder)
+
+	builder.WriteString(fmt.Sprintf("%04d", year))
+	if !isIncluded(PrecisionMonth, da.Precision) {
+		return builder.String()
+	}
+
+	if da.IsNEMA {
+		builder.WriteString(".")
+	}
+	builder.WriteString(fmt.Sprintf("%02d", month))
```

Comment:
minor nit: probably best to add a new line after this to visually separate the blocks.